### PR TITLE
SYNF-9: Add minimal long template

### PR DESCRIPTION
**Pull request by BrunoGrandePhD**
## Rationale

- The expanded template is intentionally kept simple. Each resource type is used as few times as possible. Each use should be for testing different edge cases. Edge cases should be described in the resource name. (This is included as a comment in the template as a reminder.)
- I'm avoiding use of YAML anchors and aliases since they're YAML features that aren't specific to our template format. 
- I'm avoiding a deeply nested folder structure because it doesn't really explore new edge cases. 
- I'm including the type in the names (_e.g._ `Test Project ...`) so that it's clear what's what once we work on the short format (see preview of short format below). 
- I avoided putting the minimal template into the `YAML` subfolders in `templates/` because we've decided to use YAML consistently (instead of maintaining both YAML and JSON files). 

## Questions

- Should we move the TREAT-AD templates back to the dedicated templates repository? I was thinking of keeping only the minimal long and short templates in this repo for testing purposes, and PRs that update them will also serve for discussion. 

## Preview of Short Format

Reviewing the short format will come in a future PR. I just wanted to include a preview here in to keep it in mind when suggesting tweaks to the long format, and vice versa. 

```
Test Project - With Folders:
  children:
    Test Folder - With Subfolder and One of Many:
      children:
        Test Subfolder - Empty:
    Test Folder - Empty and One of Many:

Test Project - Empty:
```

### CODE REVIEW

**Comment by thomasyu888** [comment]
> Should we move the TREAT-AD templates back to the dedicated templates repository? I was thinking of keeping only the minimal long and short templates in this repo for testing purposes, and PRs that update them will also serve for discussion.

I think we should move the TREAT-AD templates back to the templates repo.  After thinking about this some more, this repo doesn't need to have the different types of repos, just the minimal templates that can be used for testing purposes.

**Review by thomasyu888 — APPROVED** [review]

File: templates/long.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  children:
+  - name: Test Folder - With Subfolder and One of Many
+    type: Folder
+    children:
+    - name: Test Subfolder - Empty
+      type: Folder
+  - name: Test Folder - Empty and One of Many
+    type: Folder
+
+- name: Test Project - Empty
```
Comment (thomasyu888) [inline templates/long.yaml:17]: Even though we may have decided that multi project templates are out of scope, the array format allows us to do this.  I don't actually think it would be too much overhead in client code.  We are going to have to loop through top level resources anyhow

Comment (mfazza) [reply templates/long.yaml:17]: What Tom pointed out is important.  Expanding on what Tom said, reading an array of objects is a task that's already performed in expanding the short version.  Rendering anchors and merges creates an array of documents anyway.  This means that it's likely that we won't need to do anything with the object that comes from the long YAML template before feeding it to the same function.

**Comment by mfazza** [comment]
> Should we move the TREAT-AD templates back to the dedicated templates repository? I was thinking of keeping only the minimal long and short templates in this repo for testing purposes, and PRs that update them will also serve for discussion.

I think we should once we have both minimal templates.  We will still need additional testing templates though to test code coverage; so I imagine that as we work on these features, those other templates will be added to the templates folder.

File: templates/long.yaml (hunk 1 above)
Comment (BrunoGrandePhD) [reply templates/long.yaml:17]: @mattfazza: There's a lot to unpack here, so I want to break it down to make sure I understand what you're saying before merging this PR. 

Can you specify what in Tom's statement is important and why?

> Rendering anchors and merges creates an array of documents anyway. 

Did you mean "aliases" instead of "merges"? I'm not sure what you mean by "merges" otherwise. 

Also, could you clarify what you mean by "an array of documents"? Are you referring to "documents" in the [YAML spec](https://yaml.org/spec/1.2/spec.html#id2800132) sense (separated by `...` or `---`)?

Comment (mfazza) [reply templates/long.yaml:17]: > Can you specify what in Tom's statement is important and why?

Tom said "`We are going to have to loop through top level resources anyhow`".  The last implementation I remember from the short template renders an array of objects, and we were looping through those already.   On line 17, making it a list, happens to be compatible to that.  

Array of documents == [{}, {}] .  I was referring to separate objects in the YAML.  I don't think we have written different documents in the same YAML for this yet.  Merges are not applicable here - at least we haven't written template with it.  I just tend to talk about anchors (&) along with merges.  My bad if that was confusing to you.

*
> Rendering anchors and merges creates an array of documents anyway.

I just tested it, and that statement is not necessarily true.  It's implementation dependent - which is good for us because it means that we can write a template that does/doesn't do that.

Comment (BrunoGrandePhD) [reply templates/long.yaml:17]: @mattfazza: I was confused because I didn't know about merges in YAML. Could you point me to a page that describes them? A few things come up when I search for it, and I want to be sure we're talking about the same thing.

**Review by mfazza — APPROVED** [review]
Template is good, Bruno.  

Side note: I tried adding an anchor to this template, and in order to do it, the anchor needs to come as a list item before it gets aliased.  For example, this:

```yaml
- gene_folder: &gene_folder
      name: someName
  ```

has to come before its first reference.  Since the top level is a list (starts with a dash), it must also be a list item.
